src/transformers/models/superpoint/convert_superpoint_to_pytorch.py

```diff
@@ -0,0 +1,158 @@
+    rename_keys.append(("conv4b.bias", "encoder.conv4b.bias"))
+
+    # Keypoint Decoder weights
+    rename_keys.append(("convPa.weight", "keypoint_decoder.convSa.weight"))
+    rename_keys.append(("convPb.weight", "keypoint_decoder.convSb.weight"))
+    rename_keys.append(("convPa.bias", "keypoint_decoder.convSa.bias"))
+    rename_keys.append(("convPb.bias", "keypoint_decoder.convSb.bias"))
+
+    # Descriptor Decoder weights
+    rename_keys.append(("convDa.weight", "descriptor_decoder.convDa.weight"))
+    rename_keys.append(("convDb.weight", "descriptor_decoder.convDb.weight"))
+    rename_keys.append(("convDa.bias", "descriptor_decoder.convDa.bias"))
+    rename_keys.append(("convDb.bias", "descriptor_decoder.convDb.bias"))
+
+    return rename_keys
+
+
+def rename_key(dct, old, new):
+    val = dct.pop(old)
+    dct[new] = val
+
+
+def prepare_imgs():
+    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+    im1 = Image.open(requests.get(url, stream=True).raw)
+    url = "http://images.cocodataset.org/test-stuff2017/000000004016.jpg"
+    im2 = Image.open(requests.get(url, stream=True).raw)
+    return [im1, im2]
+
+
+@torch.no_grad()
+def convert_superpoint_checkpoint(checkpoint_url, pytorch_dump_folder_path, save_model, push_to_hub):
+    """
+    TODO docs
+    """
+
+    print("Downloading original model from checkpoint...")
+    config = get_superpoint_config()
+
+    original_state_dict = torch.hub.load_state_dict_from_url(checkpoint_url)
+
+    print("Converting model parameters...")
+    rename_keys = create_rename_keys(config, original_state_dict)
+    new_state_dict = original_state_dict.copy()
+    for src, dest in rename_keys:
+        rename_key(new_state_dict, src, dest)
+
+    model = SuperPointModel(config)
+    model.load_state_dict(new_state_dict)
+    model.eval()
+    print("Successfully loaded weights in the model")
+
+    preprocessor = SuperPointImageProcessor()
+    inputs = preprocessor(images=prepare_imgs(), return_tensors="pt")
+    outputs = model(**inputs)
+
+    torch.count_nonzero(outputs.mask[0])
+    expected_keypoints_shape = (2, 830, 2)
+    expected_scores_shape = (
+        2,
+        830,
+    )
+    expected_descriptors_shape = (2, 830, 256)
+
+    expected_keypoints_values = torch.tensor([[480.0, 9.0], [494.0, 9.0], [489.0, 16.0]])
+    expected_scores_values = torch.tensor([0.0064, 0.0140, 0.0595, 0.0728, 0.5170, 0.0175, 0.1523, 0.2055, 0.0336])
+    expected_descriptors_value = torch.tensor(-0.1096)
+
+    assert outputs.keypoints.shape == expected_keypoints_shape
+    assert outputs.scores.shape == expected_scores_shape
+    assert outputs.descriptors.shape == expected_descriptors_shape
+
+    assert torch.allclose(outputs.keypoints[0, :3], expected_keypoints_values, atol=1e-3)
+    assert torch.allclose(outputs.scores[0, :9], expected_scores_values, atol=1e-3)
+    assert torch.allclose(outputs.descriptors[0, 0, 0], expected_descriptors_value, atol=1e-3)
```

Comment:
testing should be optional with a flag. Tests are implemented in the testing file but if someone finetuned the original checkpoint with the original framework he won't be able to fconvert

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/378e4174f50cc63269dc88408d1d4f23facf6805)